website/templates/project/view_file.mako

```diff
@@ -2,6 +2,28 @@
 
 <div id="alertBar"></div>
 
+<%def name="resource()"><%
+        if node.get('is_registration', False):
+            nodeType = 'registration'
+        elif node.get('is_preprint', False):
+            nodeType = 'preprint'
+        elif parent_exists:
+            nodeType = 'component'
+        else:
+            nodeType = 'project'
+        return 'file for a ' + nodeType
+%>
+</%def>
+
+<%def name="public()"><%
```

Comment:
Can we define this `def` somewhere else and then just include it in these two files?